#17682 Improve eltwise binary ng test coverage

dchenTT opened the pull request:
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/17682)

### Problem description
Improve test coverage, and negative testing. 

### What's changed
Fixed bug to support sharding col_major, more than one CoreRange for core grid, 5D/ND sad path checking, and various test cases for binary and sharding.

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

dchenTT committed:
#17682 Improve eltwise binary ng test coverage

Check >=5D tensors and more pytest coverage

dchenTT committed:
#17682 Improve eltwise binary ng test coverage

Add data type variation for sharding test

dchenTT committed:
#17682 Improve eltwise binary ng test coverage

Fix bug when handling more than one core range

Add more test cases for sharding

dchenTT committed:
#17682 Improve eltwise binary ng test coverage

Add sharding orientation row and col major cases

dchenTT committed:
#17682 Improve eltwise binary ng test coverage

More coverage of col major sharded tests

dchenTT committed:
#17682 Improve eltwise binary ng test coverage

Fixed shard orientation col major.

sjameelTT reviewed (commented):
Test on post commit too. Are the negative tests all skipped?

dchenTT committed:
#17682 Improve eltwise binary ng test coverage

Allow tensor rank >= 5 but higher dimension needs to be one. Add test cases.

Ops variation test for sharding col_major

dchenTT committed:
#17682 Improve eltwise binary ng test coverage

Remove ROW_MAJOR_LAYOUT negative test  since it is supported now.

dchenTT committed:
#17682 Improve eltwise binary ng test coverage

Avoid using core coord y=7 for N300
